modules/gapi/src/backends/cpu/gcpuimgproc.cpp

```diff
@@ -136,16 +142,32 @@ GAPI_OCV_KERNEL(GCPUSobel, cv::gapi::imgproc::GSobel)
         if( borderType == cv::BORDER_CONSTANT )
         {
             cv::Mat temp_in;
-            int add = (ksize - 1) / 2;
-            cv::copyMakeBorder(in, temp_in, add, add, add, add, borderType, bordVal );
-            cv::Rect rect = cv::Rect(add, add, in.cols, in.rows);
+            cv::Rect rect = add_border(in, temp_in, ksize, borderType, bordVal);
             cv::Sobel(temp_in(rect), out, ddepth, dx, dy, ksize, scale, delta, borderType);
         }
         else
         cv::Sobel(in, out, ddepth, dx, dy, ksize, scale, delta, borderType);
     }
 };
 
+GAPI_OCV_KERNEL(GCPUSobelXY, cv::gapi::imgproc::GSobelXY)
+{
+    static void run(const cv::Mat& in, int ddepth, int dx, int dy, int ksize, double scale, double delta, int borderType,
+                    const cv::Scalar& bordVal, cv::Mat &out_dx, cv::Mat &out_dy)
+    {
+        if( borderType == cv::BORDER_CONSTANT )
+        {
+            cv::Mat temp_in;
+            cv::Rect rect = add_border(in, temp_in, ksize, borderType, bordVal);
+            cv::Sobel(temp_in(rect), out_dx, ddepth, dx, 0, ksize, scale, delta, borderType);
+            cv::Sobel(temp_in(rect), out_dy, ddepth, 0, dy, ksize, scale, delta, borderType);
+        }
+        else
+        cv::Sobel(in, out_dx, ddepth, dx, 0, ksize, scale, delta, borderType);
+        cv::Sobel(in, out_dy, ddepth, 0, dy, ksize, scale, delta, borderType);
```

Comment:
Indentation is broken a little bit. Also I believe you can avoid duplication of `cv::Sobel` calls, guess how :)

Reply:
Unfortunately the original Sobel kernel impl also duplicates this (as well as others, I believe), let's file an internal task to align this